exporter/lokiexporter/config.go

```diff
@@ -39,8 +39,11 @@ type Config struct {
 	// Labels defines how labels should be applied to log streams sent to Loki.
 	Labels LabelsConfig `mapstructure:"labels"`
 
-	// Allows you to choose the entry format in the exporter
-	Format string `mapstructure:"format"`
+	// Allows you to choose the entry format in the exporter.
+	// Deprecated: [v0.57.0] Only the JSON format will be supported in the future. If you rely on the
+	// "body" format and can't change to JSON, let us know before v0.59.0 by opening a GitHub issue
+	// and we'll work with you to find a solution.	Format string `mapstructure:"format"`
+	Format *string `mapstructure:"format"`
```

Comment:
Can this be done without an immediate breaking change to an exported struct?  Making this a pointer would enable the different messaging added here for situations where the deprecated value is provided and no value is provided, but I think that with the default config containing a value the exporter will never see a `nil` value as the default will be merged with the user-provided config before being given to the exporter.  Is the differential messaging really required? Can we just warn users who have an effective config referencing the deprecated value?

Reply:
> but I think that with the default config containing a value the exporter will never see a nil value as the default

That's true, I should have removed it from the default config. 

> Is the differential messaging really required?

I believe so: telling users that they didn't set a value and that the default will change will reduce confusion, as users seeing a generic message based on the effective value would not know where this value is coming from.